### PR TITLE
Update Travis-CI

### PR DESCRIPTION
**Pull request by ocefpaf**
Use a test matrix, test the tarball in the `install` section, remove Python 3.4 and add Python 3.6.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,8 +32,7 @@ install:
 
 script:
   - if [[ $TEST_TARGET == 'default' ]]; then
-      cp -r cc_plugin_ncei/tests/ /tmp && cd /tmp ;
-      py.test -s -rxs -v tests ;
```
Comment (ocefpaf) [inline .travis.yml]: This is a "safer" way to run tests b/c that is the condition similar to the user machine, where things are tested from a installed version. I removed it b/c there are a number of failures due to probably hardcoded paths in the tests.

I can fix those later, it is not a big deal.

**Comment by ocefpaf** [comment]
With this change the tests no longer find the `cf-standard-name-table.xml`, I believe that should come from `compliance-checker`. I'll take a closer look tomorrow or later today why that happened.

File: .travis.yml [hunk 2]
```diff
@@ -2,27 +2,39 @@ language: python
...
+    env: TEST_TARGET=default
+  - python: 3.5
+    env: TEST_TARGET=default
+  - python: 3.6
+    env: TEST_TARGET=default
+  - python: 3.6
+    env: TEST_TARGET=coding_standards
+  allow_failures:
+  - python: 3.6
+    env: TEST_TARGET=coding_standards
```
Comment (ocefpaf) [inline .travis.yml]: We should add to the test matrix a released (`default`) and latest (`allow_failures`) version of `compliance-checker` to catch errors upstream before the releases. I'll modify this PR later today to do that,

**Comment by benjwadams** [comment]
@ocefpaf, I ran into this here: https://github.com/ioos/cc-plugin-ncei/pull/40/commits/1cc24dc553b9807f9f7229834f2bb1bda9516943.   I'd make a PR on your branch, but it's easy enough to just remove the absolute path.  Let me know how you want to proceed.

**Comment by ocefpaf** [comment]
Thanks @benjwadams!

**Pull request merged**